rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -683,12 +687,15 @@ def _get_transaction_by_hash(
             self,
             web3: Optional[Web3],
             tx_hash: EVMTxHash,
-    ) -> EvmTransaction:
+    ) -> Optional[EvmTransaction]:
         if web3 is None:
             tx_data = self.etherscan.get_transaction_by_hash(tx_hash=tx_hash)
         else:
             tx_data = web3.eth.get_transaction(tx_hash)  # type: ignore
 
+        if tx_data is None:
+            return None
```

Comment:
Have you tested that both etherscan **and** querying a web3 node will return `None` for a transaction with an invalid hash here?

We should have actual tests for this.